lxd/storage/drivers/driver_btrfs_utils.go

```diff
@@ -125,14 +126,18 @@ func (d *btrfs) getSubvolumes(path string) ([]string, error) {
 
 // snapshotSubvolume creates a snapshot of the specified path at the dest supplied. If recursion is true and
 // sub volumes are found below the path then they are created at the relative location in dest.
-func (d *btrfs) snapshotSubvolume(path string, dest string, recursion bool) error {
+func (d *btrfs) snapshotSubvolume(r *revert.Reverter, path string, dest string, recursion bool) error {
```

Comment:
We've moved away from passing around revert.Reverters and instead have the function return a `revert.Hook` function that can be added to the caller's own reverter. We've found this to be easier to reason about.

Reply:
Nice pattern, like it, changed.